File: app/src/main/java/com/suraj/weatherapplication/view/activity/WeatherDetailActivity.java
package com.suraj.weatherapplication.view.activity;

import android.health.connect.datatypes.units.Temperature;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.suraj.weatherapplication.R;
import com.suraj.weatherapplication.data.CONSTANTS;
import com.suraj.weatherapplication.databinding.ActivityMainBinding;
import com.suraj.weatherapplication.databinding.ActivityWeatherDetailBinding;
import com.suraj.weatherapplication.model.WeatherData;

public class WeatherDetailActivity extends AppCompatActivity {
    private TextView cityNameTextView;
    private TextView weatherDescriptionTextView;
    private TextView temperatureTextView;
    private TextView feelsLikeTextView;
    private TextView humidityTextView;
    private TextView pressureTextView;
    ActivityWeatherDetailBinding activityWeatherDetailBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_weather_detail);

        activityWeatherDetailBinding= activityWeatherDetailBinding.inflate(getLayoutInflater());
        View view=activityWeatherDetailBinding.getRoot();
        setContentView(view);

        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,WindowManager.LayoutParams.FLAG_FULLSCREEN);


        cityNameTextView=activityWeatherDetailBinding.cityNameTextView;
        weatherDescriptionTextView=activityWeatherDetailBinding.weatherDescriptionTextView;
        temperatureTextView=activityWeatherDetailBinding.temperatureTextView;
        feelsLikeTextView=activityWeatherDetailBinding.feelsLikeTextView;
        humidityTextView=activityWeatherDetailBinding.humidityTextView;
        pressureTextView=activityWeatherDetailBinding.pressureTextView;


        WeatherData weatherData = (WeatherData) getIntent().getParcelableExtra(CONSTANTS.INTENTEXTRANAME);


        if (weatherData != null) {

            cityNameTextView.setText(weatherData.getCityName()); // City name
            weatherDescriptionTextView.setText(weatherData.getWeatherDescription().substring(0, 1).toUpperCase() + weatherData.getWeatherDescription().substring(1));// Weather description
            temperatureTextView.setText(getResources().getString(R.string.weather_detail, CONSTANTS.TEMPERATURE)+String.format("%.2f", weatherData.getTemperature())+" °C");
            feelsLikeTextView.setText(getResources().getString(R.string.weather_detail, CONSTANTS.FEELS_LIKE)+String.format("%.2f", weatherData.getFeelsLike())+" °C"); // Feels Like
            humidityTextView.setText(getResources().getString(R.string.weather_detail, CONSTANTS.HUMIDITY)+weatherData.getHumidity()+" %"); // Humidity
            pressureTextView.setText(getResources().getString(R.string.weather_detail, CONSTANTS.PRESSURE)+weatherData.getPressure()+" Pa"); // Pressure
        }
    }
}

File: app/src/main/java/com/suraj/weatherapplication/view/activity/WeatherDetailActivity.kt
package com.suraj.weatherapplication.view.activity

import android.os.Bundle
import android.provider.Settings.Global.getString
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.ui.platform.LocalContext
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.suraj.weatherapplication.R
import com.suraj.weatherapplication.data.CONSTANTS
import com.suraj.weatherapplication.model.WeatherData

class WeatherDetailActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            val weatherData = intent.getParcelableExtra<WeatherData>(CONSTANTS.INTENTEXTRANAME)
            weatherData?.let {
                WeatherDetailScreen(weatherData = it)
            }
        }
    }
}

@Composable
fun WeatherDetailScreen(weatherData: WeatherData) {
    Column(
        modifier = Modifier
            .fillMaxSize()
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .fillMaxHeight(0.4f)
                .paint(painterResource(R.drawable.weather_bg),
                    contentScale = ContentScale.Crop ),
            verticalArrangement = Arrangement.Center,
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Text(
                text = weatherData.cityName,
                fontSize = 50.sp,
                fontWeight = FontWeight.Bold,
                color = Color.Black
            )

            Text(
                text = weatherData.weatherDescription.replaceFirstChar { it.uppercase() },
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = Color.Black,
                modifier = Modifier.padding(top = 8.dp)
            )

            Spacer(modifier = Modifier.height(20.dp))
        }


        Column(
            modifier = Modifier
                .fillMaxWidth()
                .fillMaxHeight(),
            verticalArrangement = Arrangement.Center,
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Row(
                modifier = Modifier.padding(8.dp)
            ) {
                WeatherDetailItem(
                    label = LocalContext.current.getString(R.string.weather_detail, CONSTANTS.TEMPERATURE),
                    value = String.format("%.2f °C", weatherData.temperature)
                )

                WeatherDetailItem(
                    label = LocalContext.current.getString(R.string.weather_detail, CONSTANTS.FEELS_LIKE),
                    value = String.format("%.2f °C", weatherData.feelsLike)
                )
            }

            Row(
                modifier = Modifier.padding(8.dp)
            ) {
                WeatherDetailItem(
                    label = LocalContext.current.getString(R.string.weather_detail, CONSTANTS.HUMIDITY),
                    value = "${weatherData.humidity} %"
                )

                WeatherDetailItem(
                    label = LocalContext.current.getString(R.string.weather_detail, CONSTANTS.PRESSURE),
                    value = "${weatherData.pressure} Pa"
                )
            }
        }
    }
}


@Composable
fun WeatherDetailItem(label: String, value: String) {
    Box(
        modifier = Modifier
            .width(185.dp)
            .padding(10.dp)
            .height(150.dp)
            .clip(RoundedCornerShape(30.dp))
            .background(Color(0xEB0A98B4))
            .padding(16.dp),

        contentAlignment = Alignment.Center


    ) {
        Text(
            text = "$label $value",
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            color = Color.White,
            textAlign = TextAlign.Center

        )
    }
}




